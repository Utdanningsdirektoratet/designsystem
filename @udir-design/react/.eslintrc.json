{
  "extends": ["plugin:@nx/react", "../../.eslintrc.json"],
  "ignorePatterns": ["!**/*", "storybook-static"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "parserOptions": {
        "project": [
          "@udir-design/react/tsconfig.*?.json",
          "@udir-design/react/tsconfig.storybook.json"
        ]
      },
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
