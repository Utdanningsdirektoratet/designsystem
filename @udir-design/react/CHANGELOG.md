## 0.1.0-1 (2024-10-22)

build: don't bundle depenencies


## 0.1.0-0 (2024-10-22)

### ğŸš€ Features

- add Button component ([8d698eb](https://github.com/Utdanningsdirektoratet/designsystem/commit/8d698eb))
