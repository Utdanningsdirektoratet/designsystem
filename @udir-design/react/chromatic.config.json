{
  "onlyChanged": true,
  "projectId": "Project:667e8f07bf467ff4403dfe77",
  "storybookBaseDir": "@udir-design/react",
  "zip": true
}
