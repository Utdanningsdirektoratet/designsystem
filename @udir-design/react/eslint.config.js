import { FlatCompat } from '@eslint/eslintrc';
// eslint-disable-next-line @nx/enforce-module-boundaries
import baseConfig from '../../eslint.config.js';
import storybook from 'eslint-plugin-storybook';
import eslintjs from '@eslint/js';
const { configs } = eslintjs;

const compat = new FlatCompat({
  baseDirectory: import.meta.dirname,
  recommendedConfig: configs.recommended,
});

export default [
  ...baseConfig,
  ...compat.extends('plugin:@nx/react'),
  ...storybook.configs['flat/recommended'],
  {
    ignores: ['!.storybook'],
  },
  {
    files: ['**/*.ts', '**/*.tsx', '**/*.js', '**/*.jsx'],
    rules: {},
  },
  {
    files: ['**/*.ts', '**/*.tsx'],
    rules: {},
    languageOptions: {
      parserOptions: {
        project: [
          '@udir-design/react/tsconfig.*?.json',
          '@udir-design/react/tsconfig.storybook.json',
        ],
      },
    },
  },
  {
    files: ['**/*.js', '**/*.jsx'],
    rules: {},
  },
  {
    // Storybook & docs-specific overrides
    files: ['**/*.stories.{ts,tsx}', '**/{.storybook,demo,docs}/**/*.{ts,tsx}'],
    rules: {
      '@nx/enforce-module-boundaries': [
        'error',
        {
          allowCircularSelfDependency: true,
        },
      ],
    },
  },
];
