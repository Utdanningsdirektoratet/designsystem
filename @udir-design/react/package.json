{
  "name": "@udir-design/react",
  "repository": {
    "type": "git",
    "url": "github:Utdanningsdirektoratet/designsystem",
    "directory": "@udir-design/react"
  },
  "publishConfig": {
    "access": "public"
  },
  "license": "MIT",
  "version": "0.0.0-semantically-released",
  "type": "module",
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.cjs"
    },
    "./alpha": {
      "import": "./dist/alpha.js",
      "require": "./dist/alpha.cjs"
    },
    "./beta": {
      "import": "./dist/beta.js",
      "require": "./dist/beta.cjs"
    },
    "./html": {
      "types": "./dist/html.d.ts"
    },
    "./style.css": "./dist/style.css"
  },
  "files": [
    "./dist"
  ],
  "scripts": {
    "build": "vite build",
    "inject": "pnpm i",
    "lint:cycles": "pnpm madge --circular ./src/alpha.ts",
    "test": "vitest --project unit",
    "typecheck": "tsc --build",
    "dev": "storybook dev --port 6005",
    "build:storybook": "storybook build --stats-json",
    "serve:storybook": "pnpm dlx http-server ./storybook-static",
    "test:storybook": "pnpm exec playwright install --with-deps chromium && pnpm vitest --project storybook"
  },
  "dependencies": {
    "@digdir/designsystemet-react": "1.5.0",
    "@navikt/aksel-icons": "^7.30.1",
    "@udir-design/css": "workspace:*",
    "tslib": "^2.8.1"
  },
  "devDependencies": {
    "@storybook/addon-a11y": "^9.1.5",
    "@storybook/addon-docs": "^9.1.5",
    "@storybook/addon-vitest": "^9.1.5",
    "@storybook/react-vite": "^9.1.5",
    "@types/diffable-html": "^5.0.2",
    "@types/mdast": "^4.0.4",
    "@types/ramda": "^0.30.2",
    "@types/react": "^19.1.13",
    "@types/react-dom": "^19.1.9",
    "@types/remark-heading-id": "^1.0.0",
    "@vitejs/plugin-react-swc": "^4.0.1",
    "@vitest/browser": "^3.2.4",
    "@vitest/coverage-v8": "^3.2.4",
    "@vitest/ui": "^3.2.4",
    "axe-core": "^4.10.3",
    "axe-playwright": "^2.2.2",
    "chromatic": "^13.1.4",
    "clsx": "2.1.1",
    "diffable-html": "^6.0.1",
    "eslint-plugin-storybook": "^9.1.5",
    "hast-util-from-html": "^2.0.3",
    "hast-util-is-element": "^3.0.0",
    "hast-util-to-html": "^9.0.5",
    "highcharts": "^12.4.0",
    "highcharts-react-official": "^3.2.2",
    "markdown-to-jsx": "^7.7.13",
    "mdast-util-to-hast": "^13.2.0",
    "playwright": "^1.55.0",
    "ramda": "^0.31.3",
    "react": "^19.1.1",
    "react-dom": "^19.1.1",
    "react-hook-form": "^7.62.0",
    "react-style-stringify": "^1.2.0",
    "remark-gfm": "^4.0.1",
    "remark-heading-id": "^1.0.1",
    "remark-parse": "^11.0.0",
    "remark-stringify": "^11.0.0",
    "sass": "^1.92.1",
    "storybook": "^9.1.5",
    "storybook-addon-pseudo-states": "^9.1.5",
    "storybook-addon-tag-badges": "^2.0.2",
    "unified": "^11.0.5",
    "unist-util-remove": "^4.0.0",
    "unist-util-visit": "^5.0.0",
    "vite": "^7.1.5",
    "vite-plugin-dts": "^4.5.4",
    "vite-tsconfig-paths": "^5.1.4",
    "vitest": "^3.2.4"
  },
  "peerDependencies": {
    "@types/react": "^18.3.7 || ^19.0.0",
    "react": "^18.3.1 || ^19.0.0",
    "react-dom": "^18.3.1 || ^19.0.0"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    }
  }
}
