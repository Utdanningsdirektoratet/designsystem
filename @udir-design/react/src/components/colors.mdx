# Farger

Vi deler fargene i designsystemet inn i to hovedgrupper: Farger for digitale tjenester og identitetsfarger.

## Farger for digitale tjenester

Farger for digitale tjenester brukes i digitale brukergrensesnitt. De er definert som brand-farger og globale farger. Disse fargene er testet for kontrast og tilgjengelighet og skal alltid implementeres ved bruk av tokens, ikke hardkodede verdier. De gir forutsigbarhet og konsistens på tvers av komponenter og plattformer. Disse fargene brukes på for eksempel komponenter som knapper, bakgrunner, lenker, ikoner og lignende. Designteamet har definert hvordan og hvor disse fargene kan brukes:

- For mange komponenter er spesifikke fargevalg forhåndsbestemt.

- For andre komponenter og flater er det mer fleksibilitet, og designeren må ta valg ut ifra hensikt og visuell helhet.

Identitetsfargene utgjør det visuelle uttrykket til merkevaren i brukergrensesnittet og er med på å skape gjenkjennelighet og opplevelsen av et helhetlig produkt.

// Bruk og fleksibilitet - Noe om hvor man kan bruke accent, neutral, support. Noen steder har Designteamet definert valg. Andre steder er det opp til systemet/tjenesten å velge

### Brand-farger

Brand-fargene er versjoner av identitetsfargene som er tilpasset interaktive flater. På den måten opprettholder vi både merkevaregjenkjennelse og god brukeropplevelse i digitale produkter.

// Noe om de forskjellige brand-fargene - accent, neutral, support 1, support 2

### Globale farger

Globale farger er farger med funksjonell betydning og formål, og brukes til å formilde tilstand og hensikt, for info, suksess, fare og varsel. Disse brukes konsekvent i komponenter som meldingsbokser, varsler og statusindikatorer for å skape gjenkjennelighet og tydelig kommunikasjon. De har tilhørende nyanser og kontrastversjoner slik at de kan tilpasses både bakgrunn, tekst og ikoner.
Disse fargene følger vanlig konvensjonell betydning:

- **Blå** = `info`
- **Grønn** = `success`
- **Gul** = `warning`
- **Rød* = `danger`

Disse fargene er tilgjengelig i komponenter der det er relevant (for eksempel varsler, statusindikatorer og meldingskomponenter), men skal brukes med omhu.

Semantiske farger **skal ikke brukes som dekor** eller fordi de «ser fine ut». De skal kun benyttes i kontekst der innholdet har tilsvarende funksjon – som varsler, eller statusendringer.

(preview av fargetokens gruppert i brand-farger og globale farger)

## Identitetsfarger

I tillegg til fargene som er tilgjengelig som tokens for digitale brukergrensenitt, så har Udir identitetsfarger. Disse fargene brukes blant annet til illustrasjoner, malverk, pamfletter, bærenett og plakater. Alle identitetsfargene har hver sin palett, slik at de er fleksible i bruk.

## Hovedfarger

Hovedfargene til Udir er Udir-grønn og Udir-sort. Begge fargene er å finne i logoen.

## Støttefarger

Støttefargene til Udir er Udir-blå og Udir-brun. Disse fargene brukes for å gi variasjon og støtte til hovedfargene.