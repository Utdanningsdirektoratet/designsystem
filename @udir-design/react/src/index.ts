export * from './button/Button';
export * from './accordion/Accordion';
export * from './alert/Alert';
export * from './avatar/Avatar';
export * from './badge/Badge';
export * from './breadcrumbs/Breadcrumbs';
export * from './card/Card';
export * from './checkbox/Checkbox';
export * from './chip/Chip';
export * from './divider/Divider';
export * from './dropdown/Dropdown';
export * from './error-summary/ErrorSummary';
export * from './fieldset/Fieldset';
export * from './help-text/HelpText';
export * from './link/Link';
export * from './list/List';
export * from './loaders/index';
export * from './modal/Modal';
export * from './pagination/Pagination';
export * from './popover/Popover';
export * from './radio/Radio';
export * from './search/Search';
export * from './select/Select';
export * from './skip-link/SkipLink';
export * from './switch/Switch';
export * from './table/Table';
export * from './tabs/Tabs';
export * from './tag/Tag';
