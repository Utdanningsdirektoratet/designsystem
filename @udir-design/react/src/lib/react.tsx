import styles from './react.module.scss';

export const React: React.FC = () => (
  <div className={styles['container']}>
    <h1>Welcome to React!</h1>
  </div>
);

export default React;
