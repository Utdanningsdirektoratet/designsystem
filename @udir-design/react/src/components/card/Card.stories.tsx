import type { Meta, StoryObj } from '@storybook/react';
import { PlusIcon, TrashFillIcon } from '@navikt/aksel-icons';
import {
  Button,
  Card,
  Field,
  Heading,
  Label,
  Paragraph,
  Select,
  Textfield,
} from '@udir-design/react/alpha';
import { Color } from '@digdir/designsystemet-react/colors';
import { Fragment } from 'react/jsx-runtime';

const studentsImg =
  'https://images.unsplash.com/photo-1454165804606-c3d57bc86b40?q=80&w=1470&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D';

const schoolSuppliesImg =
  'https://images.unsplash.com/photo-1503676260728-1c00da094a0b?q=80&w=1644&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D';

const meta: Meta<typeof Card> = {
  component: Card,
  tags: ['alpha'],
  parameters: {
    customStyles: {
      width: '100%',
      maxWidth: 800,
      alignItems: 'center',
      display: 'grid',
      gap: 'var(--ds-size-4)',
      gridTemplateColumns: 'repeat(auto-fit, minmax(280px , 320px))',
    },
  },
};

export default meta;
type Story = StoryObj<typeof Card>;

export const Preview: Story = {
  args: {
    'data-color': 'neutral',
    children: [
      <Heading>Tittel</Heading>,
      <Paragraph>Innhold</Paragraph>,
      <Paragraph data-size="sm">Fotnote</Paragraph>,
    ],
  },
};

const VariantsMap: Record<
  string,
  { 'data-color': Color; variant: 'default' | 'tinted' }
> = {
  neutralDefault: {
    'data-color': 'neutral',
    variant: 'default',
  },
  neutralTinted: {
    'data-color': 'neutral',
    variant: 'tinted',
  },
  accentDefault: {
    'data-color': 'accent',
    variant: 'default',
  },
  accentTinted: {
    'data-color': 'accent',
    variant: 'tinted',
  },
  support1Default: {
    'data-color': 'support1',
    variant: 'default',
  },
  support1Tinted: {
    'data-color': 'support1',
    variant: 'tinted',
  },
  support2Default: {
    'data-color': 'support2',
    variant: 'default',
  },
  support2Tinted: {
    'data-color': 'support2',
    variant: 'tinted',
  },
};

export const Variants: Story = {
  render: (args) => (
    <>
      {Object.entries(VariantsMap).map(([key, value]) => (
        <Card key={key} {...args} {...value}>
          <Card.Block>
            <Paragraph>
              {Object.entries(value).map(([v, k]) => (
                <Fragment key={v}>
                  {v}: {k}
                  <br />
                </Fragment>
              ))}
            </Paragraph>
          </Card.Block>
        </Card>
      ))}
    </>
  ),
};

export const Media: Story = {
  render: (args) => (
    <>
      <Card {...args}>
        <Card.Block>
          <img
            src={schoolSuppliesImg}
            alt="Fotografi av forskjellig skoleutstyr"
          />
        </Card.Block>
        <Card.Block>
          <Heading>Skoleutstyr</Heading>
          <Paragraph>
            Fylkeskommunen skal sørge for at elevene i videregående skole får
            gratis lærebøker og andre trykte og digitale læremidler og digitalt
            utstyr.
          </Paragraph>
        </Card.Block>
      </Card>
      <Card {...args}>
        <Card.Block>
          <Heading>Skoleutstyr</Heading>
          <Paragraph>
            Fylkeskommunen skal sørge for at elevene i videregående skole får
            gratis lærebøker og andre trykte og digitale læremidler og digitalt
            utstyr.
          </Paragraph>
        </Card.Block>
        <Card.Block>
          <img
            src={schoolSuppliesImg}
            alt="Fotografi av forskjellig skoleutstyr"
          />
        </Card.Block>
      </Card>
    </>
  ),
};

export const Video: Story = {
  args: {
    'data-color': 'neutral',
    children: [
      <Card.Block>
        <iframe
          data-chromatic="ignore"
          src="https://player.vimeo.com/video/992000551?h=03734e8f67"
          width="320px"
          height="179px"
          allow="autoplay; fullscreen; picture-in-picture"
          title="30 år med digitalt innsyn"
        ></iframe>
      </Card.Block>,
      <Card.Block>
        <Heading>
          <a
            href="https://www.udir.no/regelverk-og-tilsyn/skole-og-opplaring/filmer/"
            target="_blank"
            rel="noreferrer"
          >
            Ny opplæringslov
          </a>
        </Heading>
        <Paragraph>
          Kunnskapsminister Kari Nessa Nordtun om den nye opplæringsloven.
        </Paragraph>
      </Card.Block>,
    ],
  },
};

const options = [
  { value: 'lærer', label: 'Lærer' },
  { value: 'elev', label: 'Elev' },
  { value: 'forelder', label: 'Forelder' },
];

export const Composed: Story = {
  render: (args, context) => (
    <Card {...args}>
      <Card.Block>
        <div
          style={{
            display: 'flex',
            justifyContent: 'space-between',
            alignItems: 'center',
          }}
        >
          <Heading>Rolle 1</Heading>
          <Button variant="secondary" data-color="danger" data-size="sm">
            <TrashFillIcon aria-hidden />
            Fjern
          </Button>
        </div>
      </Card.Block>
      <Card.Block>
        <Field>
          <Label>Velg rolle</Label>
          <Select id={context.id + '-role'}>
            {options.map(({ value, label }, index) => (
              <Select.Option key={index} value={value}>
                {label}
              </Select.Option>
            ))}
          </Select>
        </Field>
        <Textfield
          id={context.id + '-nationality-number'}
          label="Fødselsnummer"
        />
        <Textfield id={context.id + '-last-name'} label="Etternavn" />
      </Card.Block>
      <Card.Block>
        <Button variant="secondary" data-size="sm">
          Legg til rolle
          <PlusIcon aria-hidden />
        </Button>
      </Card.Block>
    </Card>
  ),
};

export const WithLink: Story = {
  render: (args) => (
    <>
      <Card data-color="support1" {...args}>
        <Card.Block>
          <img src={studentsImg} alt="" />
        </Card.Block>
        <Card.Block>
          <Heading>
            <a
              href="https://www.udir.no/eksamen-og-prover/eksamen/ta-fag-som-privatist/"
              target="_blank"
              rel="noopener noreferrer"
            >
              Påmelding til eksamen
            </a>
          </Heading>
          <Paragraph>
            Du må melde deg på innen 1. februar for våreksamen og 15. september
            for høsteksamen.
          </Paragraph>
          <Paragraph data-size="sm">Privatisteksamen</Paragraph>
        </Card.Block>
      </Card>
      <Card {...args} data-color="neutral">
        <Card.Block>
          <Heading>
            <a
              href="https://www.udir.no/eksamen-og-prover/eksamen/ta-fag-som-privatist/"
              target="_blank"
              rel="noopener noreferrer"
            >
              Påmelding til eksamen
            </a>
          </Heading>
          <Paragraph>
            Du må melde deg på innen 1. februar for våreksamen og 15. september
            for høsteksamen.
          </Paragraph>
          <Paragraph data-size="sm">Privatisteksamen</Paragraph>
        </Card.Block>
        <Card.Block>
          <img src={studentsImg} alt="" />
        </Card.Block>
      </Card>
    </>
  ),
};

export const Horizontal: Story = {
  parameters: {
    customStyles: {
      gridTemplateColumns: 'repeat(1fr)',
    },
  },
  args: {
    style: { display: 'grid', gridTemplateColumns: '1fr 1fr' },
    children: [
      <Card.Block>
        <Heading>Card med grid</Heading>
      </Card.Block>,
      <Card.Block>
        <Paragraph>
          Eksempel på hvordan du kan bruke et Card horisontalt ved bruk av grid.
        </Paragraph>
      </Card.Block>,
    ],
  },
};
