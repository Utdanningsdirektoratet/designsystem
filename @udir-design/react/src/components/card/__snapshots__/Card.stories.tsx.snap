// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`As Button 1`] = `
<button
  type="button"
  data-variant="default"
  data-color="support1"
>
  <div>
    <h2>
      Kort som knapp med blokker
    </h2>
  </div>
  <div>
    <p data-variant="default">
      Eksempel på kort som knapp som er delt opp med blokker
    </p>
  </div>
</button>
<button
  type="button"
  data-variant="default"
  data-color="neutral"
>
  <h2>
    Kort som knapp
  </h2>
  <p data-variant="default">
    Eksempel på kort som knapp uten blokker
  </p>
</button>
`;

exports[`As Link 1`] = `
<a
  href="https://udir.no"
  target="_blank"
  rel="noopener noreferrer"
  data-variant="default"
  data-color="support1"
>
  <div>
    <h2>
      Lenkekort med blokker
    </h2>
  </div>
  <div>
    <p data-variant="default">
      Eksempel på kort som lenke som er delt opp med blokker
    </p>
  </div>
</a>
<a
  href="https://udir.no"
  target="_blank"
  rel="noopener noreferrer"
  data-variant="default"
  data-color="neutral"
>
  <h2>
    Lenkekort
  </h2>
  <p data-variant="default">
    Eksempel på kort som lenke uten blokker
  </p>
</a>
`;

exports[`Composed 1`] = `
<div data-variant="default">
  <div>
    <div style="display: flex; justify-content: space-between; align-items: center;">
      <h2>
        Rolle 1
      </h2>
      <button
        data-variant="secondary"
        type="button"
        data-color="danger"
        data-size="sm"
      >
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="1em"
          height="1em"
          fill="none"
          viewbox="0 0 24 24"
          focusable="false"
          role="img"
          aria-hidden="true"
        >
          <path
            fill="currentColor"
            fill-rule="evenodd"
            d="M10 4.75c-.69 0-1.25.56-1.25 1.25v.25h6.5V6c0-.69-.56-1.25-1.25-1.25zm6.75 1.5V6A2.75 2.75 0 0 0 14 3.25h-4A2.75 2.75 0 0 0 7.25 6v.25H4.5a.75.75 0 0 0 0 1.5h.805l.876 11.384a1.75 1.75 0 0 0 1.745 1.616h8.148a1.75 1.75 0 0 0 1.745-1.616l.875-11.384h.806a.75.75 0 0 0 0-1.5h-2.75m-6 4.25a.75.75 0 0 0-1.5 0v6a.75.75 0 0 0 1.5 0zM14 9.75a.75.75 0 0 1 .75.75v6a.75.75 0 0 1-1.5 0v-6a.75.75 0 0 1 .75-.75"
            clip-rule="evenodd"
          >
          </path>
        </svg>
        Fjern
      </button>
    </div>
  </div>
  <div>
    <div>
      <label
        data-weight="medium"
        for="components-card--composed-role"
      >
        Velg rolle
      </label>
      <select id="components-card--composed-role">
        <option value="lærer">
          Lærer
        </option>
        <option value="elev">
          Elev
        </option>
        <option value="forelder">
          Forelder
        </option>
      </select>
    </div>
    <div>
      <label
        data-weight="medium"
        for="components-card--composed-nationality-number"
      >
        Fødselsnummer
      </label>
      <div>
        <input
          id="components-card--composed-nationality-number"
          type="text"
        >
      </div>
    </div>
    <div>
      <label
        data-weight="medium"
        for="components-card--composed-last-name"
      >
        Etternavn
      </label>
      <div>
        <input
          id="components-card--composed-last-name"
          type="text"
        >
      </div>
    </div>
  </div>
  <div>
    <button
      data-variant="secondary"
      type="button"
      data-size="sm"
    >
      Legg til rolle
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width="1em"
        height="1em"
        fill="none"
        viewbox="0 0 24 24"
        focusable="false"
        role="img"
        aria-hidden="true"
      >
        <path
          fill="currentColor"
          d="M12.75 5.5a.75.75 0 0 0-1.5 0v5.75H5.5a.75.75 0 0 0 0 1.5h5.75v5.75a.75.75 0 0 0 1.5 0v-5.75h5.75a.75.75 0 0 0 0-1.5h-5.75z"
        >
        </path>
      </svg>
    </button>
  </div>
</div>
`;

exports[`Horizontal 1`] = `
<div
  data-variant="default"
  style="display: grid; grid-template-columns: 1fr 1fr;"
>
  <div>
    <h2>
      Kort med grid
    </h2>
  </div>
  <div>
    <p data-variant="default">
      Eksempel på hvordan du kan bruke et kort horisontalt ved bruk av grid.
    </p>
  </div>
</div>
`;

exports[`Media 1`] = `
<div data-variant="default">
  <div>
    <img
      alt="skoleutstyr"
      src="https://images.unsplash.com/photo-1503676260728-1c00da094a0b?q=80&amp;w=1644&amp;auto=format&amp;fit=crop&amp;ixlib=rb-4.0.3&amp;ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
    >
  </div>
  <div>
    <h2>
      Skoleutstyr
    </h2>
    <p data-variant="default">
      Fylkeskommunen skal sørge for at elevene i videregående skole får gratis lærebøker og andre trykte og digitale læremidler og digitalt utstyr.
    </p>
  </div>
</div>
<div data-variant="default">
  <div>
    <h2>
      Skoleutstyr
    </h2>
    <p data-variant="default">
      Fylkeskommunen skal sørge for at elevene i videregående skole får gratis lærebøker og andre trykte og digitale læremidler og digitalt utstyr.
    </p>
  </div>
  <div>
    <img
      alt="skoleutstyr"
      src="https://images.unsplash.com/photo-1503676260728-1c00da094a0b?q=80&amp;w=1644&amp;auto=format&amp;fit=crop&amp;ixlib=rb-4.0.3&amp;ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
    >
  </div>
</div>
`;

exports[`Preview 1`] = `
<div
  data-variant="default"
  data-color="neutral"
>
  <h2>
    Kort
  </h2>
  <p data-variant="default">
    Innhold
  </p>
  <p
    data-variant="default"
    data-size="sm"
  >
    Fotnote
  </p>
</div>
`;

exports[`Variants 1`] = `
<div
  data-variant="default"
  data-color="neutral"
>
  <div>
    <p data-variant="default">
      data-color: neutral
      <br>
      variant: default
      <br>
    </p>
  </div>
</div>
<div
  data-variant="tinted"
  data-color="neutral"
>
  <div>
    <p data-variant="default">
      data-color: neutral
      <br>
      variant: tinted
      <br>
    </p>
  </div>
</div>
<div
  data-variant="default"
  data-color="accent"
>
  <div>
    <p data-variant="default">
      data-color: accent
      <br>
      variant: default
      <br>
    </p>
  </div>
</div>
<div
  data-variant="tinted"
  data-color="accent"
>
  <div>
    <p data-variant="default">
      data-color: accent
      <br>
      variant: tinted
      <br>
    </p>
  </div>
</div>
<div
  data-variant="default"
  data-color="support1"
>
  <div>
    <p data-variant="default">
      data-color: support1
      <br>
      variant: default
      <br>
    </p>
  </div>
</div>
<div
  data-variant="tinted"
  data-color="support1"
>
  <div>
    <p data-variant="default">
      data-color: support1
      <br>
      variant: tinted
      <br>
    </p>
  </div>
</div>
<div
  data-variant="default"
  data-color="support2"
>
  <div>
    <p data-variant="default">
      data-color: support2
      <br>
      variant: default
      <br>
    </p>
  </div>
</div>
<div
  data-variant="tinted"
  data-color="support2"
>
  <div>
    <p data-variant="default">
      data-color: support2
      <br>
      variant: tinted
      <br>
    </p>
  </div>
</div>
`;

exports[`Video 1`] = `
<div
  data-variant="default"
  data-color="neutral"
>
  <div>
    <iframe
      data-chromatic="ignore"
      src="https://player.vimeo.com/video/992000551?h=03734e8f67"
      width="320px"
      height="179px"
      allow="autoplay; fullscreen; picture-in-picture"
      title="30 år med digitalt innsyn"
    >
    </iframe>
  </div>
  <div>
    <h2>
      <a
        href="https://www.udir.no/regelverk-og-tilsyn/skole-og-opplaring/filmer/"
        target="_blank"
        rel="noreferrer"
      >
        Ny opplæringslov
      </a>
    </h2>
    <p data-variant="default">
      Kunnskapsminister Kari Nessa Nordtun om den nye opplæringsloven.
    </p>
  </div>
</div>
`;

exports[`With Link 1`] = `
<div
  data-variant="default"
  data-color="support1"
>
  <div>
    <img
      alt="studenter"
      src="https://images.unsplash.com/photo-1454165804606-c3d57bc86b40?q=80&amp;w=1470&amp;auto=format&amp;fit=crop&amp;ixlib=rb-4.0.3&amp;ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
    >
  </div>
  <div>
    <h2>
      <a
        href="https://www.udir.no/eksamen-og-prover/eksamen/ta-fag-som-privatist/"
        target="_blank"
        rel="noopener noreferrer"
      >
        Påmelding til eksamen
      </a>
    </h2>
    <p data-variant="default">
      Du må melde deg på innen 1. februar for våreksamen og 15. september for høsteksamen.
    </p>
    <p
      data-variant="default"
      data-size="sm"
    >
      Privatisteksamen
    </p>
  </div>
</div>
<div
  data-variant="default"
  data-color="neutral"
>
  <div>
    <h2>
      <a
        href="https://www.udir.no/eksamen-og-prover/eksamen/ta-fag-som-privatist/"
        target="_blank"
        rel="noopener noreferrer"
      >
        Påmelding til eksamen
      </a>
    </h2>
    <p data-variant="default">
      Du må melde deg på innen 1. februar for våreksamen og 15. september for høsteksamen.
    </p>
    <p
      data-variant="default"
      data-size="sm"
    >
      Privatisteksamen
    </p>
  </div>
  <div>
    <img
      alt="studenter"
      src="https://images.unsplash.com/photo-1454165804606-c3d57bc86b40?q=80&amp;w=1470&amp;auto=format&amp;fit=crop&amp;ixlib=rb-4.0.3&amp;ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
    >
  </div>
</div>
`;
